modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/task/CollectPartitionStatistics.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.stat.task;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.processors.affinity.AffinityTopologyVersion;
+import org.apache.ignite.internal.processors.cache.CacheGroupContext;
+import org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtLocalPartition;
+import org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtPartitionTopology;
+import org.apache.ignite.internal.processors.cache.persistence.CacheDataRow;
+import org.apache.ignite.internal.processors.query.GridQueryTypeDescriptor;
+import org.apache.ignite.internal.processors.query.h2.opt.GridH2Table;
+import org.apache.ignite.internal.processors.query.h2.opt.H2Row;
+import org.apache.ignite.internal.processors.query.stat.ColumnStatistics;
+import org.apache.ignite.internal.processors.query.stat.ColumnStatisticsCollector;
+import org.apache.ignite.internal.processors.query.stat.GatherStatisticCancelException;
+import org.apache.ignite.internal.processors.query.stat.GatherStatisticRetryException;
+import org.apache.ignite.internal.processors.query.stat.LocalStatisticsGatheringContext;
+import org.apache.ignite.internal.processors.query.stat.ObjectPartitionStatisticsImpl;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.gridgain.internal.h2.table.Column;
+
+import static org.apache.ignite.internal.processors.cache.distributed.dht.topology.GridDhtPartitionState.OWNING;
+
+/**
+ * Implementation of statistic collector.
+ */
+public class CollectPartitionStatistics implements Callable<ObjectPartitionStatisticsImpl> {
```

Comment:
@AMashenkov forced me to use gather for the statistics collection process. Are u sure about the name?